src/camera/v4l2/SDL_camera_v4l2.c

```diff
@@ -1206,10 +1074,80 @@ static int MaybeRemoveDevice(const char *path)
     }
     return 0;
 }
-#endif /* SDL_USE_LIBUDEV */
+#endif // SDL_USE_LIBUDEV
+
+
+static void V4L2_Deinitialize(void)
+{
+    for (SDL_cameralist_item *item = SDL_cameralist; item; ) {
+        SDL_cameralist_item *tmp = item->next;
+        SDL_free(item->fname);
+        SDL_free(item->bus_info);
+        SDL_free(item);
+        item = tmp;
+    }
+
+    num_cameras = 0;
```

Comment:
It's probably incorrect, but I have to add the following (to current master) to avoid a leak.
This is untested on your branch.
```suggestion
    }

#ifdef SDL_USE_LIBUDEV
    SDL_UDEV_DelCallback(capture_udev_callback);
    SDL_UDEV_Quit();
#endif

    num_cameras = 0;
```
this leak:
```
Allocation 0: 272 bytes
	0x7f82f0addb47: SDL_calloc_REAL+0x45
	0x7f82f0c2ee68: SDL_UDEV_Init+0x2f
	0x7f82f0b671de: SDL_SYS_VideoCaptureInit+0x96
	0x7f82f0b65348: SDL_VideoCaptureInit+0x22
	0x7f82f0b59e5f: SDL_VideoInit+0xa7
	0x7f82f0a2b33d: SDL_InitSubSystem_REAL+0x126
	0x7f82f0a53a87: SDL_InitSubSystem+0x19
	0x40afb8: SDLTest_CommonInit+0x108
	0x406a02: SDL_AppInit+0x58b
	0x7f82f0a86fce: SDL_InitMainCallbacks+0x66
	0x7f82f0c5e9c0: SDL_EnterAppMainCallbacks_REAL+0x43
	0x7f82f0a580b3: SDL_EnterAppMainCallbacks+0x48
	0x404f44: SDL_main+0x33
	0x7f82f0a45abf: SDL_RunApp_REAL+0x42
	0x7f82f0a4b3ed: SDL_RunApp_DEFAULT+0x37
	0x7f82f0a54a31: SDL_RunApp+0x32
	0x404f6d: main+0x27
	0x7f82f0849b8a: __libc_start_call_main+0x7a
	0x7f82f0849c4b: __libc_start_main+0x8b
	0x404a95: _start+0x25
Allocation 1: 16 bytes
	0x7f82f0addb47: SDL_calloc_REAL+0x45
	0x7f82f0c2fec0: SDL_UDEV_AddCallback+0x1b
	0x7f82f0b67207: SDL_SYS_VideoCaptureInit+0xbf
	0x7f82f0b65348: SDL_VideoCaptureInit+0x22
	0x7f82f0b59e5f: SDL_VideoInit+0xa7
	0x7f82f0a2b33d: SDL_InitSubSystem_REAL+0x126
	0x7f82f0a53a87: SDL_InitSubSystem+0x19
	0x40afb8: SDLTest_CommonInit+0x108
	0x406a02: SDL_AppInit+0x58b
	0x7f82f0a86fce: SDL_InitMainCallbacks+0x66
	0x7f82f0c5e9c0: SDL_EnterAppMainCallbacks_REAL+0x43
	0x7f82f0a580b3: SDL_EnterAppMainCallbacks+0x48
	0x404f44: SDL_main+0x33
	0x7f82f0a45abf: SDL_RunApp_REAL+0x42
	0x7f82f0a4b3ed: SDL_RunApp_DEFAULT+0x37
	0x7f82f0a54a31: SDL_RunApp+0x32
	0x404f6d: main+0x27
	0x7f82f0849b8a: __libc_start_call_main+0x7a
	0x7f82f0849c4b: __libc_start_main+0x8b
	0x404a95: _start+0x25
```